native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java

```diff
@@ -0,0 +1,260 @@
+                    properties.put(Constants.ConnectionStringProps.OTHER,
+                            miscellaneousMap.getStringValue(Constants.MiscellaneousConfig.OTHER));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.POOL_IDLE_TIMEOUT)) {
+                    properties.put(Constants.ConnectionStringProps.POOL_IDLE_TIMEOUT,
+                            miscellaneousMap.getIntValue(Constants.MiscellaneousConfig.POOL_IDLE_TIMEOUT));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.POOL_MAX_SIZE)) {
+                    properties.put(Constants.ConnectionStringProps.POOL_MAX_SIZE,
+                            miscellaneousMap.getIntValue(Constants.MiscellaneousConfig.POOL_MAX_SIZE));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.POOL_MIN_SIZE)) {
+                    properties.put(Constants.ConnectionStringProps.POOL_MIN_SIZE,
+                            miscellaneousMap.getIntValue(Constants.MiscellaneousConfig.POOL_MIN_SIZE));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.POOL_WAIT_TIME)) {
+                    properties.put(Constants.ConnectionStringProps.POOL_WAIT_TIME,
+                            miscellaneousMap.getIntValue(Constants.MiscellaneousConfig.POOL_WAIT_TIME));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.PSEUDO_COLUMNS)) {
+                    properties.put(Constants.ConnectionStringProps.PSEUDO_COLUMNS,
+                            miscellaneousMap.getStringValue(Constants.MiscellaneousConfig.PSEUDO_COLUMNS));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.QUERY_PASS_THROUGH)) {
+                    properties.put(Constants.ConnectionStringProps.QUERY_PASS_THROUGH,
+                            miscellaneousMap.getBooleanValue(Constants.MiscellaneousConfig.QUERY_PASS_THROUGH));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.RTK)) {
+                    properties.put(Constants.ConnectionStringProps.RTK,
+                            miscellaneousMap.getStringValue(Constants.MiscellaneousConfig.RTK));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.TIMEOUT)) {
+                    properties.put(Constants.ConnectionStringProps.TIMEOUT,
+                            miscellaneousMap.getIntValue(Constants.MiscellaneousConfig.TIMEOUT));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.USE_CONNECTION_POOLING)) {
+                    properties.put(Constants.ConnectionStringProps.USE_CONNECTION_POOLING,
+                            miscellaneousMap.getBooleanValue(Constants.MiscellaneousConfig.USE_CONNECTION_POOLING));
+                }
+            }
+        }
```

Comment:
We have to handle the options based on the Snowflake API requirements. Please remove these unncessary options handled specific for the CData connect domain. You can use the approach used in the following repo https://github.com/ballerina-platform/module-ballerinax-java.jdbc/blob/master/native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java.

Reply:
Updated